prerelease/action.yaml

```diff
@@ -5,67 +5,198 @@ inputs:
+
+    - id: check-pr
+      shell: bash
+      run: |
+        if [ -z "${{ steps.PR.outputs.number }}" ]; then
+          echo "pr_found=false" >> $GITHUB_OUTPUT
+          echo "has_prerelease_label=false" >> $GITHUB_OUTPUT
+        else
+          echo "pr_found=true" >> $GITHUB_OUTPUT
+          echo "has_prerelease_label=${{ contains(toJSON(fromJSON(steps.PR.outputs.pr).labels.*.name), 'prerelease') }}" >> $GITHUB_OUTPUT
+        fi
+        git branch
+        echo "branch: ${{ steps.source-vars.outputs.branch }}"
+
+    # Allow the action to download any private dependency in the go library
     - name: Authorize
-      uses: open-turo/action-git-auth@v2
+      uses: open-turo/action-git-auth@v3
       with:
         github-personal-access-token: ${{ inputs.github-token }}
+
     - name: Setup tools
       ## Installs version of golang found in .go-version
       uses: open-turo/action-setup-tools@v2
-    - name: Semantic release
-      uses: go-semantic-release/action@v1
-      id: release
+
+    - name: Prerelease
+      id: prerelease
+      uses: open-turo/actions-release/semantic-release@v5
+      if: steps.check-pr.outputs.has_prerelease_label == 'true'
       with:
+        branches: '["${{ github.event.repository.default_branch }}", {"name": "${{ steps.source-vars.outputs.branch }}","channel": "next","prerelease": "pr-${{ steps.PR.outputs.number }}.${{ github.run_number }}.${{ github.run_attempt }}"}]'
+        dry-run: ${{ inputs.create-prerelease == 'false' }}
+        extra-plugins: ${{ inputs.extra-plugins }}
         github-token: ${{ inputs.github-token }}
-    - name: Goreleaser
-      uses: goreleaser/goreleaser-action@v6
+        override-github-ref-name: ${{ steps.source-vars.outputs.branch }}
       env:
         GITHUB_TOKEN: ${{ inputs.github-token }}
-        GOVERSION: ${{ inputs.go-version }}
-      with:
-        args: release --snapshot
+    - id: vars
+      if: steps.check-pr.outputs.has_prerelease_label == 'true'
+      shell: bash
+      run: |
+        echo "version=${{ steps.prerelease.outputs.new-release-version }}" >> $GITHUB_OUTPUT
+        echo "run-url=https://github.com/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" >> "$GITHUB_OUTPUT"
+
+    - name: Set Go Version
+      shell: bash
+      run: |
+        echo "GOVERSION=$(< .go-version tr -d '\n')" >> $GITHUB_ENV
+
     - name: Docker login
-      if: inputs.push-docker-snapshot == 'true'
+      if: inputs.dockerhub-user != '' && inputs.dockerhub-password != '' && steps.release.outputs.new-release-version != ''
       uses: docker/login-action@v3
       with:
         username: ${{ inputs.docker-username }}
         password: ${{ inputs.docker-password }}
-    - name: Push Snapshot Image
-      if: inputs.push-docker-snapshot == 'true'
+
+    - name: Goreleaser
+      uses: goreleaser/goreleaser-action@v6
+      env:
+        GITHUB_TOKEN: ${{ inputs.github-token }}
+      with:
+        args: release
```

Comment:
Unless we set `create-prerelease` to true, goreleaser will fail, so maybe this doesn't need to be an input? 

This has the side effect that we will always have a draft release in Github, but that at least allows to push the compiled binaries somewhere (for the case where we need to have binaries and not a docker image)

Reply:
if it's not an input then how would we pass in the GH token?

Reply:
Oh I meant the `create-prerelease` input, not the github token, sorry